pkg/kvstore/allocator/allocator_test.go

```diff
@@ -111,6 +115,117 @@ func (s *AllocatorSuite) BenchmarkAllocate(c *C) {
 
 }
 
+func (s *AllocatorSuite) TestRunLocksGC(c *C) {
+	allocatorName := randomTestName()
+	maxID := idpool.ID(256 + c.N)
+	// FIXME: Did this previousy use allocatorName := randomTestName() ? so TestAllocatorKey(randomeTestName())
+	backend1, err := NewKVStoreBackend(allocatorName, "a", TestAllocatorKey(""))
+	c.Assert(err, IsNil)
+	c.Assert(err, IsNil)
+	allocator, err := allocator.NewAllocator(TestAllocatorKey(""), backend1, allocator.WithMax(maxID), allocator.WithoutGC())
+	c.Assert(err, IsNil)
+	shortKey := TestAllocatorKey("1;")
+	var (
+		lock1, lock2 kvstore.KVLocker
+		gotLock1     = make(chan struct{})
+		gotLock2     = make(chan struct{})
+	)
+	go func() {
+		var (
+			err error
+		)
+		lock1, err = backend1.Lock(context.Background(), shortKey)
+		c.Assert(err, IsNil)
+		close(gotLock1)
+		var client kvstore.BackendOperations
+		switch s.backend {
+		case "etcd":
+			client, _ = kvstore.NewClient(context.Background(),
+				s.backend,
+				map[string]string{
+					kvstore.EtcdAddrOption: kvstore.EtcdDummyAddress(),
+				},
+				nil,
+			)
+		case "consul":
+			client, _ = kvstore.NewClient(context.Background(),
+				s.backend,
+				map[string]string{
+					kvstore.ConsulAddrOption:   kvstore.ConsulDummyAddress(),
+					kvstore.ConsulOptionConfig: kvstore.ConsulDummyConfigFile(),
+				},
+				nil,
+			)
+		}
+		lock2, err = client.LockPath(context.Background(), allocatorName+"/locks/"+shortKey.GetKey())
+		c.Assert(err, IsNil)
+		close(gotLock2)
+	}()
+	staleLocks := map[string]kvstore.Value{}
+	staleLocks, err = allocator.RunLocksGC(staleLocks)
+	c.Assert(err, IsNil)
+	c.Assert(len(staleLocks), Equals, 0)
+
+	// Wait until lock1 is gotten.
+	c.Assert(testutils.WaitUntil(func() bool {
+		select {
+		case <-gotLock1:
+			return true
+		default:
+			return false
+		}
+	}, 5*time.Second), IsNil)
+
+	// wait until lock2 is hold
+	time.Sleep(time.Second)
```

Comment:
Why `time.Sleep()`? Why not `<-gotLock2`?

Reply:
bad comment, the lock2 won't be held, line 160 will block until lock1 is released. (2 clients can't hold the lock at the same time that's why line 160 blocks)

Reply:
>  line 160 will block until lock1 is released.

This would be useful to point out above line 160.